tests/end2end/features/thumbnail/test_thumbnaillazy.feature

```diff
@@ -0,0 +1,66 @@
+Feature: Lazy load thumbnails
+
+    Scenario: Eager loading (default behavior)
+        Given I open 10 images without leading zeros in their name
+        And I enter thumbnail mode
+        Then there should be 10 rendered thumbnails
+        When I run 5scroll right
+        Then there should be 10 rendered thumbnails
+        And the first index should be 0
+        And the last index should be 9
+
+    Scenario: Only display the selected thumbnail
+        Given I open 10 images without leading zeros in their name
+        And I enter thumbnail mode
+        Then there should be 10 rendered thumbnails
```

Comment:
I would remove this `Then`, as it is essentially tested by the previous test, no? Same for the following tests.

Reply:
The use of the `Then` there is admittedly a bit of a hack. You'll note that in the implementation of these narrative steps, there's the `wait_for_thumbnails_to_load` called at the beginning of every step. The tests don't behave unless the thumbnail rendering is finished before changing the settings.

It appears there's some kind of race condition, and I wonder if a similar race condition contributes to the bug you mention below.

Reply:
Fixed by e44a9383547eb1c4689c9a8794b45d814d914b47